src/Twig/MediaRuntime.php

```
@@ -144,6 +144,17 @@ public function path($media, string $format): string
      */
     private function getMedia($media): ?MediaInterface
     {
+        // NEXT_MAJOR: Throw exception instead and remove references to null media being accepted.
```

Maybe remove `null` from the parameter typehints?

---

Might be enough indeed, like we don't check if someone pass a boolean or an object.

---

I can remove null typehint only on 4.x, after the merge

---

Sure, but then, do we still need to throw an InvalidArgument ?
Or the method might need to be rewritten in a way we don't only check for null value but also value which is not int|string|MediaInterface

---

Anyway, we can still merge this and discuss it on the 4.x PR

---

Oh I see, since we are not relying on phpdoc and we can't typehint it (we are not on php 8) it might be better to check if the value is not accepted here? and throw a TypeError on 4.x?

---

Yes

---

Wdyt about the new implementation?